### ISSUE TITLE
Misaligned parameter highlight in schema

### ISSUE THREAD

**Issue by merqlove** [KronicDeth/intellij-elixir#621]
Renders with pink somewhere...

![screenshot 2017-02-15 23 53 29](https://cloud.githubusercontent.com/assets/18354/22995004/0e02a67e-f3da-11e6-8187-4520ed7a246b.png)

**Comment by KronicDeth** [comment]
Since the pink color is for parameters, I'm assuming there's some parameter elsewhere in the file that is *not* being highlighted.  Are you using any variables above the `embedded_schema` call in the `defmodule` body?

**Comment by merqlove** [comment]
Yeah, here smh like this at the top:
```
defmodule InFactApi.ExpertiseQuery do
  use InFactApi.Web, :id_model

  defmodule NoIterationError do
    defexception message: "Iteration not found"
  end

  alias InFactApi.{ExpertiseArticle, Repo, Imports.Factor, EctoHelper}

  @required_fields []
  @optional_fields [:names, :iteration, :state, :slop]
  @assoc_params prepare_params(~w(factor_1_id factor_2_id factor_1 factor_2))

  use EctoStateMachine,
    states: [:empty, :and, :or, :error],
    initial: :empty,
    events: [
      [
        name:     :make_empty,
        from:     [:and, :or, :error],
        to:       :empty
      ], [
        name:     :make_and,
        from:     [:empty, :or, :error],
        to:       :and
      ], [
        name:     :make_or,
        from:     [:empty, :and, :error],
        to:       :or
      ], [
        name:     :state_error,
        from:     [:empty, :and, :or],
        to:       :error
      ]
    ],
    repo: Repo

  embedded_schema do
    belongs_to :factor_1, Factor
    belongs_to :factor_2, Factor

    embeds_many :articles, ExpertiseArticle, on_replace: :delete

    field :names, {:array, :string}, default: []
    field :iteration, :integer, default: 1
    field :slop, :integer

    field :state, :string, default: "and"
  end
end
```

**Comment by KronicDeth** [comment]
@merqlove, pasting the following in a scratch tab does not trigger the mis-coloring for me.  It has to be some interaction with the rest of your project (like resolving references):

```
defmodule InFactApi.ExpertiseQuery do
  use InFactApi.Web, :id_model

  defmodule NoIterationError do
    defexception message: "Iteration not found"
  end

  alias InFactApi.{ExpertiseArticle, Repo, Imports.Factor, EctoHelper}

  @required_fields []
  @optional_fields [:names, :iteration, :state, :slop]
  @assoc_params prepare_params(~w(factor_1_id factor_2_id factor_1 factor_2))

  use EctoStateMachine,
    states: [:empty, :and, :or, :error],
    initial: :empty,
    events: [
      [
        name:     :make_empty,
        from:     [:and, :or, :error],
        to:       :empty
      ], [
        name:     :make_and,
        from:     [:empty, :or, :error],
        to:       :and
      ], [
        name:     :make_or,
        from:     [:empty, :and, :error],
        to:       :or
      ], [
        name:     :state_error,
        from:     [:empty, :and, :or],
        to:       :error
      ]
    ],
    repo: Repo

  embedded_schema do
    belongs_to :factor_1, Factor
    belongs_to :factor_2, Factor

    embeds_many :articles, ExpertiseArticle, on_replace: :delete

    field :names, {:array, :string}, default: []
    field :iteration, :integer, default: 1
    field :slop, :integer

    field :state, :string, default: "and"
  end

  def changeset(query, %{__struct__: _} = params) do
    changeset(query, Map.from_struct(params))
  end
  def changeset(query, %{} = params) do
    {instance_params, assoc_params} = EctoHelper.cleanup_params(params, @assoc_params)
  end
end
```

![screen shot 2017-02-22 at 9 59 18 pm](https://cloud.githubusercontent.com/assets/298259/23244357/33b3b3ca-f94a-11e6-805e-2037456b3300.png)

**Comment by KronicDeth** [comment]
This may be related to #626, in which case it would also be fixed by #627.

**Comment by merqlove** [comment]
Yeah, I will try it as some nightly available
